mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,400 @@
+    {
+        RemoveStickybomb(true);
+        return;
+    }
+
+    if (m_bFizzle)
+    {
+        g_pEffects->Sparks(GetAbsOrigin());
+        RemoveStickybomb(true);
+        return;
+    }
+}
+
+void CMomStickybomb::Explode(trace_t *pTrace, CBaseEntity *pOther)
+{
+    if (CNoGrenadesZone::IsInsideNoGrenadesZone(this))
+    {
+        RemoveStickybomb(true);
+        return;
+    }
+
+    // Make invisible
+    SetModelName(NULL_STRING);
+    SetSolid(SOLID_NONE);
+    m_takedamage = DAMAGE_NO;
+
+    // Pull out a bit
+    if (!CloseEnough(pTrace->fraction, 1.0f))
+    {
+        SetAbsOrigin(pTrace->endpos + (pTrace->plane.normal * 1.0f));
+    }
+
+    CBaseEntity *pOwner = GetOwnerEntity();
+
+    float flDamage = GetDamage();
+    float flRadius = GetRadius();
+
+    // Explosion effect on client
+    Vector vecOrigin = GetAbsOrigin();
+    CPVSFilter filter(vecOrigin);
+    CWeaponID iWeaponID = WEAPON_STICKYLAUNCHER;
+
+    if (UseImpactNormal())
+    {
+            TE_TFExplosion(filter, 0.0f, vecOrigin, GetImpactNormal(), iWeaponID, -1);
+    }
+    else
+    {
+            TE_TFExplosion(filter, 0.0f, vecOrigin, pTrace->plane.normal, iWeaponID, -1);
+    }
+
+    Vector vecReported = GetThrower() ? GetThrower()->GetAbsOrigin() : vec3_origin;
+
+    // Damage
+    CTakeDamageInfo info(this, pOwner, vec3_origin, vecOrigin, flDamage, GetDamageType(), 0, &vecReported);
+    RadiusDamage(info, vecOrigin, flRadius, CLASS_NONE, nullptr);
+
+    m_hOwner = nullptr;
+
+    if (pOther && !pOther->IsPlayer())
+    {
+        UTIL_DecalTrace(pTrace, "Scorch");
+    }
+
+    SetThink(&CMomStickybomb::SUB_Remove);
+    SetTouch(nullptr);
+
+    UTIL_Remove(this);
+}
+
+void CMomStickybomb::Fizzle() { m_bFizzle = true; }
+
+void CMomStickybomb::VPhysicsCollision(int index, gamevcollisionevent_t *pEvent)
+{
+    BaseClass::VPhysicsCollision(index, pEvent);
+
+    int otherIndex = !index;
+    CBaseEntity *pHitEntity = pEvent->pEntities[otherIndex];
+
+    if (!pHitEntity)
+        return;
+
+    // Handle hitting skybox (bounce off).
+    surfacedata_t *pprops = physprops->GetSurfaceData(pEvent->surfaceProps[otherIndex]);
+    if (pprops->game.material == 'X')
+    {
+        return;
+    }
+
+    bool bIsDynamicProp = (NULL != dynamic_cast<CDynamicProp *>(pHitEntity));
+
+    // Stickybombs stick to the world when they touch it
+    if (pHitEntity && (pHitEntity->IsWorld() || bIsDynamicProp) && gpGlobals->curtime > 0)
+    {
+        m_bTouched = true;
+        VPhysicsGetObject()->EnableMotion(false);
+
+        // Save impact data for explosions.
+        m_bUseImpactNormal = true;
+        pEvent->pInternalData->GetSurfaceNormal(m_vecImpactNormal);
+        m_vecImpactNormal.Negate();
+    }
+}
+
+void CMomStickybomb::StickybombTouch(CBaseEntity *pOther)
+{
+    Assert(pOther);
+
+    // Don't touch triggers
+    if (pOther->IsSolidFlagSet(FSOLID_TRIGGER | FSOLID_VOLUME_CONTENTS))
+        return;
```

Comment:
It looks like this check (along with a missing `!pOther->IsSolid() ||` before) was used in the original pipebomb code for the grenade launcher's pipes since those are meant to explode on impact with a solid entity, so this should be able to be safely removed.